node/core/candidate-validation/src/lib.rs

```diff
@@ -539,7 +557,7 @@ fn perform_basic_checks(
 		return Err(InvalidCandidate::PoVHashMismatch)
 	}
 
-	if *validation_code_hash != candidate.validation_code_hash {
+	if let Some(false) = validation_code_hash.map(|hash| *hash == candidate.validation_code_hash) {
```

Comment:
This should be expressable as, right? That'd be a little more concise.

```suggestion
	if Some(&candidate.validation_code_hash) == validation_code_hash {
```

Reply:
Not quite, we want to check whether it's *not* equal to the expected one.
But that's a good catch, after the upgrade the validation code hash is always present so this place can be fixed